command/connect/proxy/proxy.go

```diff
@@ -244,20 +244,15 @@ func LookupProxyIDForSidecar(client *api.Client, sidecarFor string) (string, err
 	return proxyIDs[0], nil
 }
 
-// LookupGatewayProxyID finds the gateway service registered with the local
-// agent if any and returns its service ID. It will return an ID if and only if
-// there is exactly one gateway of this kind registered to the agent.
+// LookupGatewayProxy finds the gateway service registered with the local
+// agent. If exactly one gateway exists it will be returned, otherwise an error
+// is returned.
 func LookupGatewayProxy(client *api.Client, kind api.ServiceKind) (*api.AgentService, error) {
 	svcs, err := client.Agent().ServicesWithFilter(fmt.Sprintf("Kind == `%s`", kind))
 	if err != nil {
 		return nil, fmt.Errorf("Failed looking up %s instances: %v", kind, err)
 	}
 
-	var proxyIDs []string
-	for _, svc := range svcs {
-		proxyIDs = append(proxyIDs, svc.ID)
-	}
```

Comment:
Should this be used? I wasn't sure how to correct this, so I removed it for now.

Reply:
Unclear if it matters right now, since the overall method `LookupGatewayProxy` isn't used by anything at the moment.

Reply:
The builtin-proxy (`consul connect proxy`) doesn't support mesh gateways currently.

Reply:
Oh ya, this is in the builtin-proxy package, but it is called from `consul connect envoy`, which does support them. I guess it might be appropriate to move this to the envoy package, which is the single caller.